src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateCopy = {
+    ...state,
+  };
+  const previousVersions = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
```

Comment:
```suggestion
   const { type, extraData, keysToRemove } = actions[i]
    switch (actions[i].type) {
```
consider to destructure action